pkg/minio/minio_test.go

```diff
@@ -234,6 +238,33 @@ func TestFGetObject(t *testing.T) {
 	assert.NilError(t, err)
 }
 
+func TestNewClientAndFGetObjectWithProxy(t *testing.T) {
+	// start proxy
+	proxyListener, err := net.Listen("tcp", ":0")
```

Comment:
looking at the code of this library it looks bad to me, because it allocates a random port and then releases it. this makes it possible for a concurrent syscall for binding to that port to steal this (not really) allocated port. this way we allocate the random port atomically

Reply:
```go
// GetFreePort asks the kernel for a free open port that is ready to use.
func GetFreePort() (int, error) {
	addr, err := net.ResolveTCPAddr("tcp", "localhost:0")
	if err != nil {
		return 0, err
	}

	l, err := net.ListenTCP("tcp", addr)
	if err != nil {
		return 0, err
	}
	defer l.Close()
	return l.Addr().(*net.TCPAddr).Port, nil
}
```